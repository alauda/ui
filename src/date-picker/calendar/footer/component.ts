import {
  ChangeDetectionStrategy,
  Component,
  EventEmitter,
  Input,
  Output,
  TemplateRef,
  ViewEncapsulation,
} from '@angular/core';

@Component({
  selector: 'aui-calendar-footer',
  templateUrl: './template.html',
  styleUrls: ['./style.scss'],
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CalendarFooterComponent {
  @Input()
  clearable = true;

  @Input()
  clearText: string;

  @Input()
  customAction: TemplateRef<any>;

  @Output()
  confirm = new EventEmitter<void>();

  @Output()
  clear = new EventEmitter<void>();
}
