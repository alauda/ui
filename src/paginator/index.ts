export * from './paginator.component';
export * from './paginator.module';
export * from './paginator-intl';
