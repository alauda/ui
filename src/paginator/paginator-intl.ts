import { Injectable, Optional, SkipSelf } from '@angular/core';
import { Subject } from 'rxjs';

/**
 * To modify the labels and text displayed, create a new instance of MatPaginatorIntl and
 * include it in a custom provider
 */
@Injectable()
export class PaginatorIntl {
  /**
   * Stream that emits whenever the labels here are changed. Use this to notify
   * components if the labels have changed after initialization.
   */
  readonly changes: Subject<void> = new Subject<void>();

  /** A label for the page size selector. */
  itemsPerPageLabel = 'perPage';

  jumperLabelPrefix = 'Goto';
  jumperLabelSuffix = '';

  getTotalLabel = (length: number) => `Total ${length}`;
}

// eslint-disable-next-line sonar/function-name
export function PAGINATOR_INTL_PROVIDER_FACTORY(parentIntl: PaginatorIntl) {
  return parentIntl || new PaginatorIntl();
}

export const PAGINATOR_INTL_PROVIDER = {
  // If there is already an PaginatorIntl available, use that. Otherwise, provide a new one.
  provide: PaginatorIntl,
  deps: [[new Optional(), new SkipSelf(), PaginatorIntl]],
  useFactory: PAGINATOR_INTL_PROVIDER_FACTORY,
};
