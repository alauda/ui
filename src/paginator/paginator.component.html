<div [class]="bem.block()">
  <div *ngFor="let layout of getLayoutArr()" [class]="bem.element('layout')">
    <div *ngIf="layout === 'pager'" [class]="bem.element('pager')">
      <button
        [ngClass]="bem.element('navigator')"
        aui-button
        [disabled]="disabled || currentPage <= 1"
        (click)="navigate(-1)"
      >
        <aui-icon icon="angle_left"></aui-icon>
      </button>
      <button
        [ngClass]="bem.element('page')"
        [aui-button]="currentPage === 1 ? 'primary' : 'default'"
        [plain]="!(currentPage === 1)"
        [disabled]="disabled"
        (click)="jumpTo(1)"
      >
        1
      </button>
      <button
        *ngIf="pageCount > 7 && currentPage > 4"
        [ngClass]="bem.element('fast-navigator')"
        aui-button
        [disabled]="disabled"
        (click)="navigate(-5)"
      >
        <aui-icon icon="ellipsis_h_s"></aui-icon>
        <aui-icon icon="angle_double_left"></aui-icon>
      </button>
      <button
        *ngFor="let page of getCenterPages(); trackBy: trackByIndex"
        [ngClass]="bem.element('page')"
        [aui-button]="currentPage === page ? 'primary' : 'default'"
        [plain]="!(currentPage === page)"
        [disabled]="disabled"
        (click)="jumpTo(page)"
      >
        {{ page }}
      </button>
      <button
        *ngIf="pageCount > 7 && currentPage < pageCount - 3"
        [ngClass]="bem.element('fast-navigator')"
        aui-button
        [disabled]="disabled"
        (click)="navigate(5)"
      >
        <aui-icon icon="ellipsis_h_s"></aui-icon>
        <aui-icon icon="angle_double_right"></aui-icon>
      </button>
      <button
        *ngIf="pageCount > 1"
        [ngClass]="bem.element('page')"
        [aui-button]="currentPage === pageCount ? 'primary' : 'default'"
        [plain]="!(currentPage === pageCount)"
        [disabled]="disabled"
        (click)="jumpTo(pageCount)"
      >
        {{ pageCount }}
      </button>
      <button
        aui-button
        [ngClass]="bem.element('navigator')"
        [disabled]="disabled || currentPage >= pageCount"
        (click)="navigate(1)"
      >
        <aui-icon icon="angle_right"></aui-icon>
      </button>
    </div>
    <div *ngIf="layout === 'total'" [class]="bem.element('total')">
      {{ intl.getTotalLabel(total) }}
    </div>
    <div *ngIf="layout === 'sizes'" [class]="bem.element('sizes')">
      <aui-select
        [disabled]="disabled"
        [value]="pageSize"
        [filterable]="false"
        (valueChange)="changePageSize($event)"
      >
        <aui-option *ngFor="let size of pageSizeOptions" [value]="size">
          {{ size }}
        </aui-option>
      </aui-select>
      <span>{{ intl.itemsPerPageLabel }}</span>
    </div>
    <div *ngIf="layout === 'jumper'" [class]="bem.element('jumper')">
      <span *ngIf="intl.jumperLabelPrefix">{{ intl.jumperLabelPrefix }}</span>
      <input
        aui-input
        type="number"
        min="1"
        [max]="pageCount"
        [value]="currentPage"
        [disabled]="disabled"
        (keydown)="onKeydown($event)"
      />
      <span *ngIf="intl.jumperLabelSuffix">{{ intl.jumperLabelSuffix }}</span>
    </div>
    <div *ngIf="layout === 'content'" [class]="bem.element('content')">
      <ng-content></ng-content>
    </div>
  </div>
</div>
