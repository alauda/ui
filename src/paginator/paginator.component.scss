@import '../theme/var';
@import '../theme/mixin';

$block: 'aui-paginator';
$input-width: 80px;

.#{$block} {
  display: inline-flex;
  align-items: center;
  color: use-rgb(n-1);

  .aui-button {
    width: use-var(line-height-xl);
    height: use-var(line-height-xl);
    border: unset;

    &--default {
      background-color: transparent;
    }

    .aui-button__content {
      padding: unset;
    }
  }

  &__fast-navigator {
    &[hidden] {
      display: none;
    }

    .aui-icon {
      margin: 0;
    }

    aui-icon:last-child {
      display: none;
    }

    &:hover aui-icon {
      &:first-child {
        display: none;
      }

      &:last-child {
        display: inline;
      }
    }
  }

  &__sizes {
    display: inline-flex;
    align-items: center;

    .aui-select {
      width: $input-width;
      margin-right: use-var(spacing-s);
    }
  }

  &__jumper {
    .aui-input {
      width: $input-width;
    }

    span + input {
      margin-left: use-var(spacing-m);
    }

    input + span {
      margin-left: use-var(spacing-s);
    }
  }

  &__total {
    font-size: use-var(font-size-m);
  }

  &__navigator.aui-button {
    background-color: use-rgb(n-8);

    aui-icon {
      font-size: use-var(font-size-l);
    }

    &:hover {
      background-color: use-rgb(p-6);
    }
  }

  &__pager {
    border: 1px solid use-rgb(n-7);
    padding: 3px;
    border-radius: use-var(border-radius-m);
    background-color: use-rgb(main-bg);
  }

  &__layout {
    display: inline-block;

    + .#{$block}__layout {
      margin-left: use-var(spacing-l);
    }
  }
}
