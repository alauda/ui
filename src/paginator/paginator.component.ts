import {
  ChangeDetectionStrategy,
  ChangeDetectorRef,
  Component,
  EventEmitter,
  Input,
  OnDestroy,
  Output,
  ViewEncapsulation,
} from '@angular/core';
import { Subscription } from 'rxjs';

import { Bem, buildBem } from '../utils';

import { PaginatorIntl } from './paginator-intl';

/**
 * Change event object that is emitted when the user selects a
 * different page size or navigates to another page.
 * @deprecated use Output events directly instead
 */
export class PageEvent {
  /** The current page index. */
  pageIndex: number;

  /**
   * Index of the page that was selected previously.
   */
  previousPageIndex: number;

  /** The current page size */
  pageSize: number;

  /** The current total number of items being paged */
  length: number;
}

@Component({
  selector: 'aui-paginator',
  templateUrl: 'paginator.component.html',
  styleUrls: ['paginator.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None,
  preserveWhitespaces: false,
})
export class PaginatorComponent implements OnDestroy {
  bem: Bem = buildBem('aui-paginator');

  @Input()
  get pageIndex() {
    return this.currentPage - 1;
  }

  set pageIndex(val) {
    this.currentPage = val + 1;
  }

  @Input()
  currentPage = 1;

  @Input()
  pageSize = 20;

  @Input()
  pageSizeOptions: number[] = [20, 50, 100];

  @Input()
  total = 0;

  /**
   * @deprecated use `total` instead
   */
  @Input()
  get length() {
    return this.total;
  }

  set length(val) {
    this.total = val;
  }

  @Input()
  layout: string | string[] = 'total,pager,sizes';

  @Input()
  disabled = false;

  @Output()
  readonly currentPageChange = new EventEmitter<number>();

  @Output()
  readonly pageSizeChange = new EventEmitter<number>();

  private readonly intlChangeSub: Subscription;

  get pageCount() {
    return Math.ceil(this.total / this.pageSize);
  }

  constructor(
    public intl: PaginatorIntl,
    private readonly cdr: ChangeDetectorRef,
  ) {
    this.intlChangeSub = intl.changes.subscribe(() => this.cdr.markForCheck());
  }

  ngOnDestroy() {
    this.intlChangeSub.unsubscribe();
  }

  getLayoutArr() {
    return typeof this.layout === 'string'
      ? this.layout.split(',').map(item => item.trim())
      : this.layout;
  }

  getCenterPages() {
    const start = Math.min(
      Math.max(0, this.currentPage - 4),
      Math.max(0, this.pageCount - 7),
    );
    return Array.from({ length: Math.max(this.pageCount - 2, 0) })
      .fill(null)
      .map((_, index) => index + 2)
      .slice(start, start + 5);
  }

  jumpTo(page: number) {
    if (page === this.currentPage) {
      return;
    }
    this.currentPageChange.emit(page);
  }

  navigate(step: number) {
    this.jumpTo(Math.min(Math.max(this.currentPage + step, 1), this.pageCount));
  }

  changePageSize(size: number) {
    if (size === this.pageSize) {
      return;
    }
    this.pageSizeChange.emit(size);
  }

  onKeydown(event: KeyboardEvent) {
    if (event.key === 'Enter') {
      const inputEl = event.target as HTMLInputElement;
      const nextPage = parseInt(inputEl.value, 10);
      if (nextPage > 0 && nextPage <= this.pageCount) {
        this.jumpTo(nextPage);
      } else {
        inputEl.value = this.currentPage + '';
      }
      event.stopPropagation();
      event.preventDefault();
    }
  }

  trackByIndex(index: number) {
    return index;
  }
}
