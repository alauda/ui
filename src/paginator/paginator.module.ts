import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { ButtonModule } from '../button';
import { IconModule } from '../icon';
import { InputModule } from '../input';
import { SelectModule } from '../select';

import { PAGINATOR_INTL_PROVIDER } from './paginator-intl';
import { PaginatorComponent } from './paginator.component';

@NgModule({
  imports: [CommonModule, ButtonModule, IconModule, SelectModule, InputModule],
  declarations: [PaginatorComponent],
  exports: [PaginatorComponent],
  providers: [PAGINATOR_INTL_PROVIDER],
})
export class PaginatorModule {}
