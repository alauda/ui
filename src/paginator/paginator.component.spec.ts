import { Component, DebugElement, ViewChild } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { PaginatorComponent } from './paginator.component';

import { PaginatorModule } from '.';

describe('PaginatorComponent', () => {
  let fixture: ComponentFixture<TestComponent>;
  let ins: TestComponent;
  let debugEl: DebugElement;
  let el: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [PaginatorModule],
      declarations: [TestComponent],
    });

    fixture = TestBed.createComponent(TestComponent);
    fixture.detectChanges();
    ins = fixture.componentInstance;
    debugEl = fixture.debugElement.query(By.css('aui-paginator'));
    el = debugEl.nativeElement;
  });

  it('should render current template', () => {
    expect(fixture).toMatchSnapshot();

    ins.disabled = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should navigate working correctly', () => {
    const navBtns = el.querySelectorAll('.aui-paginator__navigator');
    const getFastNavBtn = (i = 0) =>
      el.querySelectorAll('.aui-paginator__fast-navigator').item(i);
    const getCurrentPageBtn = () =>
      el.querySelector('.aui-paginator__page.aui-button--primary');

    expect(getCurrentPageBtn().innerHTML).toContain('1');

    navBtns.item(1).dispatchEvent(new Event('click'));
    fixture.detectChanges();
    expect(getCurrentPageBtn().innerHTML).toContain('2');
    expect(ins.onPageChange.mock.calls[0][0]).toEqual(2);

    navBtns.item(0).dispatchEvent(new Event('click'));
    fixture.detectChanges();
    expect(getCurrentPageBtn().innerHTML).toContain('1');
    expect(ins.onPageChange.mock.calls[1][0]).toEqual(1);

    getFastNavBtn().dispatchEvent(new Event('click'));
    fixture.detectChanges();
    expect(getCurrentPageBtn().innerHTML).toContain('6');
    expect(ins.onPageChange.mock.calls[2][0]).toEqual(6);

    getFastNavBtn().dispatchEvent(new Event('click'));
    fixture.detectChanges();
    expect(getCurrentPageBtn().innerHTML).toContain('1');
    expect(ins.onPageChange.mock.calls[3][0]).toEqual(1);
  });

  it('should jump to correct page when click page button', () => {
    const pageBtns = el.querySelectorAll('.aui-paginator__page');
    const lastPageBtn = pageBtns.item(pageBtns.length - 1);
    const pageCount = parseInt(
      lastPageBtn.querySelector('.aui-button__content').innerHTML.trim(),
      10,
    );

    lastPageBtn.dispatchEvent(new Event('click'));
    fixture.detectChanges();
    expect(lastPageBtn.className).toContain('aui-button--primary');
    expect(ins.onPageChange.mock.calls[0][0]).toEqual(pageCount);
  });

  it('should jumper working correctly', () => {
    const jumperEl: HTMLInputElement = el.querySelector(
      '.aui-paginator__jumper .aui-input',
    );
    jumperEl.value = '10';
    jumperEl.dispatchEvent(new KeyboardEvent('keydown', { key: 'Enter' }));
    fixture.detectChanges();
    expect(
      el.querySelector('.aui-paginator__page.aui-button--primary').innerHTML,
    ).toContain('10');

    jumperEl.value = '1000';
    jumperEl.dispatchEvent(new KeyboardEvent('keydown', { key: 'Enter' }));
    fixture.detectChanges();
    expect(
      el.querySelector('.aui-paginator__page.aui-button--primary').innerHTML,
    ).toContain('10');
    expect(jumperEl.value).toEqual('10');
  });

  it('should change page size correctly', () => {
    ins.paginator.changePageSize(100);
    fixture.detectChanges();
    expect(ins.onPageSizeChange.mock.calls[0][0]).toEqual(100);
    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  template: `
    <aui-paginator
      [(currentPage)]="currentPage"
      [(pageSize)]="pageSize"
      [total]="total"
      [pageSizeOptions]="pageSizeOptions"
      [layout]="layout"
      [disabled]="disabled"
      (currentPageChange)="onPageChange($event)"
      (pageSizeChange)="onPageSizeChange($event)"
    >
      custom content
    </aui-paginator>
  `,
})
class TestComponent {
  currentPage = 1;
  pageSize = 20;
  total = 350;
  pageSizeOptions = [20, 50, 100];
  layout = ['total', 'pager', 'sizes', 'jumper', 'custom content'];
  disabled = false;

  onPageChange = jest.fn();
  onPageSizeChange = jest.fn();

  @ViewChild(PaginatorComponent, { static: true })
  paginator: PaginatorComponent;
}
