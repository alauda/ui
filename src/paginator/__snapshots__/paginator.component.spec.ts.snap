// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatorComponent should change page size correctly 1`] = `
<ng-component
  currentPage={[Function Number]}
  disabled="false"
  layout={[Function Array]}
  onPageChange={[Function Function]}
  onPageSizeChange={[Function Function]}
  pageSize={[Function Number]}
  pageSizeOptions={[Function Array]}
  paginator={[Function PaginatorComponent]}
  total={[Function Number]}
>
  <aui-paginator>
    <div
      class="aui-paginator"
    >
      
      <div
        class="aui-paginator__layout"
      >
        
        
        <div
          class="aui-paginator__total"
        >
           Total 350 
        </div>
        
        
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        <div
          class="aui-paginator__pager"
        >
          <button
            aui-button=""
            class="aui-button aui-button--default aui-button--medium aui-paginator__navigator"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
              <aui-icon
                icon="angle_left"
              >
                
                
                <svg
                  class="aui-icon aui-icon-angle_left"
                >
                  <use
                    xlink:href="#aui-icon-angle_left"
                  />
                </svg>
                
                
              </aui-icon>
            </span>
            
          </button>
          <button
            class="aui-button aui-button--medium aui-paginator__page aui-button--primary"
          >
            <span
              class="aui-button__content"
            >
               1 
            </span>
            
          </button>
          
          
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
          >
            <span
              class="aui-button__content"
            >
               2 
            </span>
            
          </button>
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
          >
            <span
              class="aui-button__content"
            >
               3 
            </span>
            
          </button>
          
          
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
          >
            <span
              class="aui-button__content"
            >
               4 
            </span>
            
          </button>
          <button
            aui-button=""
            class="aui-button aui-button--default aui-button--medium aui-paginator__navigator"
          >
            <span
              class="aui-button__content"
            >
              <aui-icon
                icon="angle_right"
              >
                
                
                <svg
                  class="aui-icon aui-icon-angle_right"
                >
                  <use
                    xlink:href="#aui-icon-angle_right"
                  />
                </svg>
                
                
              </aui-icon>
            </span>
            
          </button>
        </div>
        
        
        
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        
        
        <div
          class="aui-paginator__sizes"
        >
          <aui-select>
            <div
              auitooltipposition="bottom start"
              auitooltiptrigger="click"
              auitooltiptype="plain"
              class="aui-select aui-select--medium"
            >
              <aui-input-group>
                <div
                  class="aui-input-group aui-input-group--medium hasSuffix"
                >
                  <div
                    class="aui-input-group__inner-wrapper"
                  >
                    
                    <div
                      class="aui-input-group__input-wrapper"
                    >
                      
                      <input
                        aui-input=""
                        autocomplete="off"
                        class="aui-select__input aui-input aui-input--medium"
                        placeholder=""
                        readonly=""
                      />
                      
                      <div
                        class="aui-select__label-container aui-input aui-input--medium"
                      >
                        <div
                          class="aui-select__label"
                        >
                          
                          
                          
                          100
                          
                        </div>
                      </div>
                      
                      <span
                        class="aui-input-group__suffix"
                      >
                        
                        <aui-icon
                          auiinputsuffix=""
                          class="aui-select__indicator"
                        >
                          
                          
                          <svg
                            class="aui-icon aui-icon-caret_down_s"
                          >
                            <use
                              xlink:href="#aui-icon-caret_down_s"
                            />
                          </svg>
                          
                          
                        </aui-icon>
                      </span>
                    </div>
                    
                  </div>
                </div>
              </aui-input-group>
            </div>
            
          </aui-select>
          <span>
            perPage
          </span>
        </div>
        
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        
        
        
        <div
          class="aui-paginator__jumper"
        >
          
          <span>
            Goto
          </span>
          <input
            aui-input=""
            class="aui-input aui-input--medium"
            max="4"
            min="1"
            type="number"
          />
          
        </div>
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        
        
        
        
      </div>
    </div>
  </aui-paginator>
</ng-component>
`;

exports[`PaginatorComponent should render current template 1`] = `
<ng-component
  currentPage={[Function Number]}
  disabled="false"
  layout={[Function Array]}
  onPageChange={[Function Function]}
  onPageSizeChange={[Function Function]}
  pageSize={[Function Number]}
  pageSizeOptions={[Function Array]}
  paginator={[Function PaginatorComponent]}
  total={[Function Number]}
>
  <aui-paginator>
    <div
      class="aui-paginator"
    >
      
      <div
        class="aui-paginator__layout"
      >
        
        
        <div
          class="aui-paginator__total"
        >
           Total 350 
        </div>
        
        
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        <div
          class="aui-paginator__pager"
        >
          <button
            aui-button=""
            class="aui-button aui-button--default aui-button--medium aui-paginator__navigator"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
              <aui-icon
                icon="angle_left"
              >
                
                
                <svg
                  class="aui-icon aui-icon-angle_left"
                >
                  <use
                    xlink:href="#aui-icon-angle_left"
                  />
                </svg>
                
                
              </aui-icon>
            </span>
            
          </button>
          <button
            class="aui-button aui-button--medium aui-paginator__page aui-button--primary"
          >
            <span
              class="aui-button__content"
            >
               1 
            </span>
            
          </button>
          
          
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
          >
            <span
              class="aui-button__content"
            >
               2 
            </span>
            
          </button>
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
          >
            <span
              class="aui-button__content"
            >
               3 
            </span>
            
          </button>
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
          >
            <span
              class="aui-button__content"
            >
               4 
            </span>
            
          </button>
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
          >
            <span
              class="aui-button__content"
            >
               5 
            </span>
            
          </button>
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
          >
            <span
              class="aui-button__content"
            >
               6 
            </span>
            
          </button>
          
          <button
            aui-button=""
            class="aui-button aui-button--default aui-button--medium aui-paginator__fast-navigator"
          >
            <span
              class="aui-button__content"
            >
              <aui-icon
                icon="ellipsis_h_s"
              >
                
                
                <svg
                  class="aui-icon aui-icon-ellipsis_h_s"
                >
                  <use
                    xlink:href="#aui-icon-ellipsis_h_s"
                  />
                </svg>
                
                
              </aui-icon>
              <aui-icon
                icon="angle_double_right"
              >
                
                
                <svg
                  class="aui-icon aui-icon-angle_double_right"
                >
                  <use
                    xlink:href="#aui-icon-angle_double_right"
                  />
                </svg>
                
                
              </aui-icon>
            </span>
            
          </button>
          
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
          >
            <span
              class="aui-button__content"
            >
               18 
            </span>
            
          </button>
          <button
            aui-button=""
            class="aui-button aui-button--default aui-button--medium aui-paginator__navigator"
          >
            <span
              class="aui-button__content"
            >
              <aui-icon
                icon="angle_right"
              >
                
                
                <svg
                  class="aui-icon aui-icon-angle_right"
                >
                  <use
                    xlink:href="#aui-icon-angle_right"
                  />
                </svg>
                
                
              </aui-icon>
            </span>
            
          </button>
        </div>
        
        
        
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        
        
        <div
          class="aui-paginator__sizes"
        >
          <aui-select>
            <div
              auitooltipposition="bottom start"
              auitooltiptrigger="click"
              auitooltiptype="plain"
              class="aui-select aui-select--medium"
            >
              <aui-input-group>
                <div
                  class="aui-input-group aui-input-group--medium hasSuffix"
                >
                  <div
                    class="aui-input-group__inner-wrapper"
                  >
                    
                    <div
                      class="aui-input-group__input-wrapper"
                    >
                      
                      <input
                        aui-input=""
                        autocomplete="off"
                        class="aui-select__input aui-input aui-input--medium"
                        placeholder=""
                        readonly=""
                      />
                      
                      <div
                        class="aui-select__label-container aui-input aui-input--medium"
                      >
                        <div
                          class="aui-select__label"
                        >
                          
                          
                          
                          20
                          
                        </div>
                      </div>
                      
                      <span
                        class="aui-input-group__suffix"
                      >
                        
                        <aui-icon
                          auiinputsuffix=""
                          class="aui-select__indicator"
                        >
                          
                          
                          <svg
                            class="aui-icon aui-icon-caret_down_s"
                          >
                            <use
                              xlink:href="#aui-icon-caret_down_s"
                            />
                          </svg>
                          
                          
                        </aui-icon>
                      </span>
                    </div>
                    
                  </div>
                </div>
              </aui-input-group>
            </div>
            
          </aui-select>
          <span>
            perPage
          </span>
        </div>
        
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        
        
        
        <div
          class="aui-paginator__jumper"
        >
          
          <span>
            Goto
          </span>
          <input
            aui-input=""
            class="aui-input aui-input--medium"
            max="18"
            min="1"
            type="number"
          />
          
        </div>
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        
        
        
        
      </div>
    </div>
  </aui-paginator>
</ng-component>
`;

exports[`PaginatorComponent should render current template 2`] = `
<ng-component
  currentPage={[Function Number]}
  disabled={[Function Boolean]}
  layout={[Function Array]}
  onPageChange={[Function Function]}
  onPageSizeChange={[Function Function]}
  pageSize={[Function Number]}
  pageSizeOptions={[Function Array]}
  paginator={[Function PaginatorComponent]}
  total={[Function Number]}
>
  <aui-paginator>
    <div
      class="aui-paginator"
    >
      
      <div
        class="aui-paginator__layout"
      >
        
        
        <div
          class="aui-paginator__total"
        >
           Total 350 
        </div>
        
        
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        <div
          class="aui-paginator__pager"
        >
          <button
            aui-button=""
            class="aui-button aui-button--default aui-button--medium aui-paginator__navigator"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
              <aui-icon
                icon="angle_left"
              >
                
                
                <svg
                  class="aui-icon aui-icon-angle_left"
                >
                  <use
                    xlink:href="#aui-icon-angle_left"
                  />
                </svg>
                
                
              </aui-icon>
            </span>
            
          </button>
          <button
            class="aui-button aui-button--medium aui-paginator__page aui-button--primary"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
               1 
            </span>
            
          </button>
          
          
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
               2 
            </span>
            
          </button>
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
               3 
            </span>
            
          </button>
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
               4 
            </span>
            
          </button>
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
               5 
            </span>
            
          </button>
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
               6 
            </span>
            
          </button>
          
          <button
            aui-button=""
            class="aui-button aui-button--default aui-button--medium aui-paginator__fast-navigator"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
              <aui-icon
                icon="ellipsis_h_s"
              >
                
                
                <svg
                  class="aui-icon aui-icon-ellipsis_h_s"
                >
                  <use
                    xlink:href="#aui-icon-ellipsis_h_s"
                  />
                </svg>
                
                
              </aui-icon>
              <aui-icon
                icon="angle_double_right"
              >
                
                
                <svg
                  class="aui-icon aui-icon-angle_double_right"
                >
                  <use
                    xlink:href="#aui-icon-angle_double_right"
                  />
                </svg>
                
                
              </aui-icon>
            </span>
            
          </button>
          
          <button
            class="aui-button aui-button--default aui-button--medium aui-paginator__page isPlain"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
               18 
            </span>
            
          </button>
          <button
            aui-button=""
            class="aui-button aui-button--default aui-button--medium aui-paginator__navigator"
            disabled=""
          >
            <span
              class="aui-button__content"
            >
              <aui-icon
                icon="angle_right"
              >
                
                
                <svg
                  class="aui-icon aui-icon-angle_right"
                >
                  <use
                    xlink:href="#aui-icon-angle_right"
                  />
                </svg>
                
                
              </aui-icon>
            </span>
            
          </button>
        </div>
        
        
        
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        
        
        <div
          class="aui-paginator__sizes"
        >
          <aui-select>
            <div
              auitooltipposition="bottom start"
              auitooltiptrigger="click"
              auitooltiptype="plain"
              class="aui-select aui-select--medium isDisabled"
            >
              <aui-input-group>
                <div
                  class="aui-input-group aui-input-group--medium hasSuffix isDisabled"
                >
                  <div
                    class="aui-input-group__inner-wrapper"
                  >
                    
                    <div
                      class="aui-input-group__input-wrapper"
                    >
                      
                      <input
                        aui-input=""
                        autocomplete="off"
                        class="aui-select__input aui-input aui-input--medium"
                        disabled=""
                        placeholder=""
                        readonly=""
                      />
                      
                      <div
                        class="aui-select__label-container aui-input aui-input--medium"
                        disabled="true"
                      >
                        <div
                          class="aui-select__label"
                        >
                          
                          
                          
                          20
                          
                        </div>
                      </div>
                      
                      <span
                        class="aui-input-group__suffix"
                      >
                        
                        <aui-icon
                          auiinputsuffix=""
                          class="aui-select__indicator"
                        >
                          
                          
                          <svg
                            class="aui-icon aui-icon-caret_down_s"
                          >
                            <use
                              xlink:href="#aui-icon-caret_down_s"
                            />
                          </svg>
                          
                          
                        </aui-icon>
                      </span>
                    </div>
                    
                  </div>
                </div>
              </aui-input-group>
            </div>
            
          </aui-select>
          <span>
            perPage
          </span>
        </div>
        
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        
        
        
        <div
          class="aui-paginator__jumper"
        >
          
          <span>
            Goto
          </span>
          <input
            aui-input=""
            class="aui-input aui-input--medium"
            disabled=""
            max="18"
            min="1"
            type="number"
          />
          
        </div>
        
      </div>
      <div
        class="aui-paginator__layout"
      >
        
        
        
        
        
      </div>
    </div>
  </aui-paginator>
</ng-component>
`;
